openedx/core/djangoapps/programs/utils.py

```diff
@@ -216,11 +220,18 @@ def progress(self, programs=None, count_only=True):
                 else:
                     not_started.append(course)
 
+            grades = {}
+            for run in self.course_run_ids:
+                course = get_course_with_access(self.user, 'load', CourseKey.from_string(run))
+                grade = self.course_grade_factory.read(self.user, course)
+                grades[run] = grade.percent
+
             progress.append({
                 'uuid': program_copy['uuid'],
                 'completed': len(completed) if count_only else completed,
                 'in_progress': len(in_progress) if count_only else in_progress,
                 'not_started': len(not_started) if count_only else not_started,
+                'grades': grades,
```

Comment:
Should there be a Python related test for this change?

Reply:
I'm still not seeing a test for this.

Also, does course_grade_factory return a 0 for courses you haven't started?  In other words, it returns a grade for everything?  Or should we only be requesting for "completed"?

Reply:
It does return a grade for everything. I figured the updated `test_utils.py` tests do cover this change unless you want something else?

Reply:
Sort of.  If there is a simple way to change the grade from 0.0 for one of the course runs for a test it might be a better test.  See if that is easy enough to do.